nav2_amcl/src/amcl_node.cpp

```diff
@@ -628,7 +633,7 @@ AmclNode::laserReceived(sensor_msgs::msg::LaserScan::ConstSharedPtr laser_scan)
   if (!getOdomPose(latest_odom_pose_, pose.v[0], pose.v[1], pose.v[2],
     laser_scan->header.stamp, base_frame_id_))
   {
-    RCLCPP_ERROR(get_logger(), "Couldn't determine robot's pose associated with laser scan");
+    RCLCPP_DEBUG(get_logger(), "Couldn't determine robot's pose associated with laser scan");
```

Comment:
This laser callback is called at the laser frequency - which is typically between 5 and 40 hz - but I get your point.

The [original AMCL code](https://github.com/ros-planning/navigation/blob/melodic-devel/amcl/src/amcl_node.cpp#L1133) does this. While I hate the argument `do it because that's how its always been`, I think its valuable to have it pop up alot when there's an issue because frankly if you can't get the position in odom, there's a _great deal_ of other issues at play that AMCL is a good sentinel for identifying.

Reply:
I agree a THROTTLE log would help here. The problems right now are partly related to the fact we don't have the geometry2 message filters yet, so we're failing transforms all the time. Basically we don't have any message syncing yet. I can add this to that issue to revert this when message filtering is fixed. Also, this whole callback is going to be refactored (we have an issue for that too). I'll add links to those to make sure this get's revisited.

Reply:
I added links and comments in the other issues to make sure these get revisited.